src/scales/scale.time.js

```diff
@@ -562,6 +562,37 @@ module.exports = function(Chart) {
 			me._minorFormat = formats[unit];
 			me._majorFormat = formats[majorUnit];
 			me._table = buildLookupTable(me._timestamps.data, min, max, options.distribution);
+			me._leftOffset = 0;
+			me._rightOffset = 0;
+
+			if (options.offset && ticks.length) {
```

Comment:
I think the `buildTicks` method is already a bit too complex, can we move that logic in a separate external method:

```javascript
me._table = buildLookupTable(me._timestamps.data, min, max, options.distribution);
me._offsets = computeOffsets(table, ticks, options);   // return {left: ..., right: ...}

return ticksFromTimestamps(ticks, majorUnit);
```